### ISSUE TITLE
agent can not reconnect to the collector when the  oap server have only one instance and  has been restarted

### ISSUE THREAD

**Issue by yefei14131** [apache/skywalking#3466]
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [ ] Question or discussion
- [x] Bug
- [ ] Requirement
- [ ] Feature or performance improvement

___
### Bug
- Which version of SkyWalking, OS and JRE?
version: v6.3.0
- Which company or project?
hualala.com
- What happen?
agent can not reconnect to the collector when the  oap server only one instance and  has been restarted.

source code in class org.apache.skywalking.apm.agent.core.remote.GRPCChannelManager:
```bash
try {
    int index = Math.abs(random.nextInt()) % grpcServers.size();
    if (index != selectedIdx) {
        selectedIdx = index;

        server = grpcServers.get(index);
        String[] ipAndPort = server.split(":");

        if (managedChannel != null) {
            managedChannel.shutdownNow();
        }

        managedChannel = GRPCChannel.newBuilder(ipAndPort[0], Integer.parseInt(ipAndPort[1]))
            .addManagedChannelBuilder(new StandardChannelBuilder())
            .addManagedChannelBuilder(new TLSChannelBuilder())
            .addChannelDecorator(new AgentIDDecorator())
            .addChannelDecorator(new AuthenticationDecorator())
            .build();

        notify(GRPCChannelStatus.CONNECTED);
        reconnect = false;
    } else if (managedChannel.isConnected()) {
        // Reconnect to the same server is automatically done by GRPC,
        // therefore we are responsible to check the connectivity and
        // set the state and notify listeners
        notify(GRPCChannelStatus.CONNECTED);
        reconnect = false;
    }

    return;
} catch (Throwable t) {
    logger.error(t, "Create channel to {} fail.", server);
}
```

the line 102 in class org.apache.skywalking.apm.agent.core.remote.GRPCChannelManager should be channge to :
```bash
if (index != selectedIdx || !managedChannel.isConnected())
```

**Comment by kezhenxu94** [comment]
Duplicate of #3366

**Comment by kezhenxu94** [comment]
The bug should be fixed in 6.3.0, #3181 and #3240  @yefei14131
